Framework/PythonInterface/plugins/algorithms/EnggFitPeaks.py

```diff
@@ -264,24 +265,22 @@ def _fit_single_peak(self, expected_center, initial_params, wks, wks_index):
         peak.setParameter('I', intensity)
 
         # Fit using predicted window and a proper function with approximated initial values
-        fit_alg = self.createChildAlgorithm('Fit')
         fit_function = 'name=LinearBackground;{0}'.format(peak)
-        fit_alg.setProperty('Function', fit_function)
-        fit_alg.setProperty('InputWorkspace', wks)
-        fit_alg.setProperty('WorkspaceIndex', wks_index)
-        fit_alg.setProperty('CreateOutput', True)
-
         (startx, endx) = self._estimate_start_end_fitting_range(center, width)
-        fit_alg.setProperty('StartX', startx)
-        fit_alg.setProperty('EndX', endx)
         self.log().debug("Fitting for peak expected in (d-spacing): {0}, Fitting peak function: "
                          "{1}, with startx: {2}, endx: {3}".
                          format(expected_center, fit_function, startx, endx))
-        fit_alg.execute()
-        param_table = fit_alg.getProperty('OutputParameters').value
-        chi_over_dof = fit_alg.getProperty('OutputChi2overDoF').value
 
-        return param_table, chi_over_dof
+        fit_output = Fit(
+            Function=fit_function,
+            InputWorkspace=wks,
+            WorkspaceIndex=wks_index,
+            CreateOutput=True,
+            StartX=startx,
+            EndX=endx
+        )
+
+        return fit_output.OutputParameters, fit_output.OutputChi2overDoF
```

Comment:
This does work nicely, however leaves `Fit` output workspaces in the ADS that were not there in the earlier solution. If this additional output was not intended, then perhaps the `Fit` algorithm should also have `StoreInADS=False` .